native/src/components/ExportEventButton.tsx

```diff
@@ -27,45 +28,51 @@ const ExportEventButton = ({ event }: ExportEventButtonType): ReactElement => {
-      // can't open a specific event but at a specific time
-      const appleRefDate = DateTime.fromISO('2001-01-01').toSeconds()
-      const secondsSinceRefDate = event.date.startDate.toSeconds() - appleRefDate
-      Linking.openURL(`calshow:${secondsSinceRefDate}`)
-    } else if (Platform.OS === 'android') {
-      Linking.openURL(`content://com.android.calendar/events/${id}`)
-    }
-  }
-
   const exportEventToCalendar = async (calendarId: string | undefined): Promise<void> => {
     try {
+      let startDate = event.date.startDate.toUTC().toString()
+      let endDate = event.date.endDate.toUTC().toString()
+      const allDay = event.date.allDay
+      if (Platform.OS === 'android' && allDay) {
+        // If allDay is set to true, Android demands that the time has a midnight boundary.
+        // The endDate we receive from the CMS for allDay events is always at 23:59:00.
+        startDate = event.date.startDate.toFormat("yyyy-LL-dd'T'00:00:00.000'Z'")
+        endDate = event.date.endDate.plus({ minutes: 1 }).toFormat("yyyy-LL-dd'T'00:00:00.000'Z'")
+      }
+
       const id = await RNCalendarEvents.saveEvent(event.title, {
-        startDate: event.date.startDate.toISO(),
-        endDate: event.date.endDate.toISO(),
-        allDay: event.date.allDay,
-        location: event.location?.fullAddress,
-        description: event.excerpt,
+        startDate,
+        endDate,
+        allDay,
         calendarId,
+        location: event.location?.fullAddress,
+        description: event.excerpt, // Android
+        notes: event.excerpt, // iOS
       })
-      setEventExported(true)
-      showSnackbar({
-        text: t('added'),
-        positiveAction: {
-          label: t('goToCalendar'),
-          onPress: () => openCalendarApp(event, id),
-        },
-      })
-    } catch {
+      if (id) {
+        setEventExported(true)
+        showSnackbar({
+          text: t('added'),
+        })
+      } else {
+        showSnackbar({ text: 'generalError' })
+      }
+    } catch (e) {
       showSnackbar({ text: 'generalError' })
+      reportError(e)
     }
   }
 
   const checkCalendarsAndExportEvent = async (): Promise<void> => {
-    const checkedPermission = await RNCalendarEvents.checkPermissions()
-    const permission =
-      checkedPermission === 'undetermined' ? await RNCalendarEvents.requestPermissions() : checkedPermission
-    if (permission === 'denied' || permission === 'restricted') {
+    const iosPermission = PERMISSIONS.IOS.CALENDARS
+    const androidPermissions = [PERMISSIONS.ANDROID.READ_CALENDAR, PERMISSIONS.ANDROID.WRITE_CALENDAR]
+    const checkedPermission =
+      Platform.OS === 'ios' ? await check(iosPermission) : await checkMultiple(androidPermissions)
+    const requestPermission = () =>
+      Platform.OS === 'ios' ? request(iosPermission) : requestMultiple(androidPermissions)
+
+    const permission = checkedPermission === 'unavailable' ? await requestPermission() : checkedPermission
```

Comment:
```suggestion
    const permission = await requestPermission()
```
This check handles the library itself afaik. (Only tested on android)

Reply:
You're right, very cool!